code-challenges.test.js

```diff
@@ -14,39 +14,184 @@
+// return the coded message using the "finalMessage" variable
+
+// const codedMessage = (strings) => {
+//     let string = strings.toLowerCase()
+//     let finalMessage = ""
+//     for (let i = 0; i < string.length; i++) {
+//         if (string[i] === "a") {
+//             finalMessage += "4"
+//         } else if (string[i] === "e") {
+//             finalMessage += "3"
+//         } else if (string[i] === "i") {
+//             finalMessage += "1"
+//         } else if (string[i] === "o") {
+//             finalMessage += "0"
+//         } else {
+//             finalMessage += strings[i]
+//         }
+//    }
+//    return finalMessage
+
+// }
+
+// Test Passed
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+// Snapshots:   0 total
+// Time:        0.134 s, estimated 1 s
+
+
 // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
 // a) Create a test with expects statement using the variable provided.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
-
-const letterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const letterE = "e"
+// describe ("words", () => {
+//     it ("will take in an array and a single letter as a parameter and return an array of all the words with that particular letter", () => {
+//         expect(words(fruitArray,letterA)).toEqual(["Mango", "Apricot", "Peach"])
+//         expect(words(fruitArray,letterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+        
+        
+//     })
+//   })
+
+//   // Good Failure
+//   // Test Suites: 1 failed, 1 total
+// // Tests:       1 failed, 1 total
+// // Snapshots:   0 total
+// // Time:        0.17 s, estimated 1 s
+
+// const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+
+// const letterA = "a"
+// // Expected output: ["Mango", "Apricot", "Peach"]
+// const letterE = "e"
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
 // b) Create the function that makes the test pass.
 
+// Process
+// Create a function called "words" that takes in an array and a letter as parameters
+// Create an empty array that will store our final outcome
+// Create a for loop that will loop through the array 
+// Convert the letter parameter to lower case using the ".toLowerCase()" method
+// Convert the current word to lower case
+// Create a conditional statement that checks for the index of the letter in the word
+// If the word contains the letter then it will be pushed into the new array
+// return "finalArray"
+
+
+// const words = (array1,letter1) => {
+//     let finalArray = [] 
+//     for (let i = 0; i<array1.length; i++) {
+//         let letter2 = letter1.toLowerCase()
+//         let word = array1[i].toLowerCase()
+//         if (word.indexOf(letter2)!== -1) {
+//             finalArray.push(array1[i])
+//         }
+//     } return finalArray
+// }
```

Comment:
I like the use of `.indexOf() !== -1` to see if the word has the letter or not, but don't forget about the `.includes()` built in method! Similarly to question 1, how can you refactor this?